priv/scim.abnf

```diff
@@ -2,27 +2,29 @@
 ; https://www.rfc-editor.org/rfc/rfc7644#section-4
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-FILTER    = attrExp / logExp / valuePath / *1"not" "(" FILTER ")"
+FILTER    = (attrExp / valuePath / [NOT SP] "(" FILTER ")") [SP AND-OR SP FILTER]
 
 valuePath = attrPath "[" valFilter "]"
             ; FILTER uses sub-attributes of a parent attrPath
 
-valFilter = attrExp / logExp / *1"not" "(" valFilter ")"
+valFilter = (attrExp / [NOT SP] "(" valFilter ")") [SP AND-OR SP valFilter]
 
 attrExp   = (attrPath SP "pr") /
             (attrPath SP compareOp SP compValue)
 
-logExp    = FILTER SP ("and" / "or") SP FILTER
-
```

Comment:
Fixes missing space in `not (...)`

Flattens filter expression to remove infinite recursion

Extracts `NOT` and `AND-OR` for better customization